common/persistence/executionStore.go

```diff
@@ -406,7 +406,7 @@ func (m *executionManagerImpl) SerializeWorkflowMutation(
 	}
 
 	var serializedNewBufferedEvents *serialization.DataBlob
-	if input.NewBufferedEvents != nil {
+	if input.NewBufferedEvents != nil && len(input.NewBufferedEvents) > 0 {
```

Comment:
I would just leave the second part:
```suggestion
	if len(input.NewBufferedEvents) > 0 {
```
From [golang wiki](https://github.com/golang/go/wiki/CodeReviewComments#declaring-empty-slices) I learnt that slice should be `nil` every time they are empty but `len(slice)` is a right away to check if it is actually empty or not cause it covers both cases.

Reply:
Ah good catch! I've relied on this elsewhere, but just didn't click here when I wrote this.